src/compiler/transformers/module/module.ts

```diff
@@ -820,6 +831,14 @@ namespace ts {
             let statements: Statement[];
             let variables: VariableDeclaration[];
             let expressions: Expression[];
+
+            const parseTreeNode = getParseTreeNode(node);

```

Comment:
So just setting on in `createExportExpression` won't be enough because only VariableStatement with bindingPattern will call into the function. 

Also we can't just check for `currentModuleInfo.exportEquals === false`
because we will be issue extra __esModule even though it doesn't need one

For example:
```ts
    export enum SignatureFlags {
        None = 0,
        IsIndexer = 1,
        IsStringIndexer = 1 << 1,
        IsNumberIndexer = 1 << 2,
    }
```
will get `__esModule` as well